lib/src/carry_save_multiplier.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// carry_save_multiplier.dart
+// Implementation of pipeline multiplier module.
+//
+// 2023 February 21
+// Author: Max Korbel <max.korbel@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+
+/// A datatype class that hold result from [FullAdder].
+class FullAdderResult {
```

Comment:
Let's ditch this `FullAdderResult` object and just have `sum` and `cOut` directly as outputs (`get`ters) of the `FullAdder` module.